src/handlers/swap_handlers.ts

```diff
@@ -30,7 +30,55 @@ export class SwapHandlers {
         this._swapService = swapService;
     }
     public async getSwapQuoteAsync(req: express.Request, res: express.Response): Promise<void> {
-        // parse query params
+        res.status(HttpStatus.OK).send(
+            await this._calculateSwapQuoteAsync(parseGetSwapQuoteRequestParams(req), req.header('0x-api-key')),
+        );
+    }
+    // tslint:disable-next-line:prefer-function-over-method
+    public async getSwapTokensAsync(_req: express.Request, res: express.Response): Promise<void> {
+        const tokens = TokenMetadatasForChains.map(tm => ({
+            symbol: tm.symbol,
+            address: tm.tokenAddresses[CHAIN_ID],
+            name: tm.name,
+            decimals: tm.decimals,
+        }));
+        const filteredTokens = tokens.filter(t => t.address !== NULL_ADDRESS);
+        res.status(HttpStatus.OK).send({ records: filteredTokens });
+    }
+    // tslint:disable-next-line:prefer-function-over-method
+    public async getSwapPriceAsync(req: express.Request, res: express.Response): Promise<void> {
+        const params = parseGetSwapQuoteRequestParams(req);
+        if (params.rfqt === undefined) {
+            params.rfqt = {
+                intentOnFilling: false,
+            };
+        }
+        params.skipValidation = true;
+        const quote = await this._calculateSwapQuoteAsync(params, req.header('0x-api-key'));
+        const { price, value, gasPrice, gas, protocolFee, buyAmount, sellAmount } = quote;
```

Comment:
I know it's a change to the spec, but I think it's actually a hard requirement that we return `sources` here.  Would you mind adding that?

IMHO I think we need this so wallet UXs can show the order sources to the user before they review their order

Reply:
definitely helpful for an implementor to show the breakdown

Reply:
Addressed in https://github.com/0xProject/0x-api/pull/172/commits/cc3b686cc78a09ba48ca818fd73ec3cace73928c